### PR TITLE
Properly pass on `loadBackendConfig`'s `additionalConfig` option

### LINKED ISSUE #17380: 🐛 Bug Report: Unable to use additional configuration sources at startup
### 📜 Description

Version https://github.com/backstage/backstage/releases/tag/v1.13.0-next.2

https://github.com/backstage/backstage/pull/16643 introduced a new feature that should allow additional sources of configuration to be provided on startup (e.g. secrets from AWS Secret Manager).

However, there's a type mismatch in the signatures of `loadBackendConfig`  in `packages/backend-common/src/config.ts` and `loadBackendConfig` in `packages/backend-app-api/src/config/config.ts`.

One `loadBackendConfig` expects a non-array `AppConfig`
https://github.com/backstage/backstage/blob/32bf9ff91457184e075dc4ee578c0e753bc6e2c9/packages/backend-common/src/config.ts#L37
while the other expects an array `AppConfig[]`
https://github.com/backstage/backstage/blob/32bf9ff91457184e075dc4ee578c0e753bc6e2c9/packages/backend-app-api/src/config/config.ts#L74

This prevents me from using this new functionality, unfortunately. 

### 👍 Expected behavior

When I pass a static AppConfig to `loadBackendConfig` at startup

```
async function main() {
	const appConfig: AppConfig = {
		context: '/path/to/config',
		data: {
		  foo: 'bar',
		}
	  };

	const config = await loadBackendConfig({
		argv: process.argv,
		logger: getRootLogger(),
		additionalConfig: appConfig
	});

	console.log(config.getString('foo'));

```

I expect to see `bar` in the console log when I run backstage locally

### 👎 Actual Behavior with Screenshots

In the current state, Backstage fails to startup because 
`[1] Backend failed to start up, Error: Missing required config value at 'foo'`,

### 👟 Reproduction steps

1. Pass an instance of AppConfig to `loadBackendConfig` during backend initialization
3. Try to log a value off that AppConfig after the initialization
4. Start Backstage locally
5. Check logs and see if it reports an issue or a value 

See `Expected behavior` for code samples

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by freben**
Fixes #17380

### CODE REVIEW

File: packages/backend-common/src/config.ts [hunk 1]
```diff
@@ -40,7 +40,13 @@ export async function loadBackendConfig(options: {
...
     logger: options.logger,
   });
-  const { config } = await newLoadBackendConfig(options);
+  const { config } = await newLoadBackendConfig({
+    argv: options.argv,
+    remote: options.remote,
+    ...(options.additionalConfig
+      ? { additionalConfigs: [options.additionalConfig] }
+      : {}),
+  });
```
Comment (ilya-kochetkov-cko) [inline packages/backend-common/src/config.ts:49]: Hi, is it possible to change `additionalConfig?: AppConfig` to `additionalConfigs?: AppConfig[]` on the line 37 to enable passing **multiple** additional configuration sources when bootstrapping backstage?

Such a change should also make it possible to use the original line if I don't miss anything here.
`const { config } = await newLoadBackendConfig(options);`

Thank you 🙏

Comment (freben) [reply packages/backend-common/src/config.ts:49]: Not really, that would be a breaking change. I could make it `AppConfig | AppConfig[]` to make it backwards compatible, but not sure that it's worth it - would you have a real use of that at this point or did you mention it just for symmetry's sake?

Comment (Rugvip) [reply packages/backend-common/src/config.ts:49]: Iirc it's not in mainline yet so fine to break, best double check tho

Comment (ilya-kochetkov-cko) [reply packages/backend-common/src/config.ts:49]: Given that this change has not been released yet (scheduled for Tuesday, 18 April), will this count as a breaking change? I'm sorry for lacking knowledge of your development process, this is my first time contributing to Backstage 👀 

At this point, given that the top-level signature (backend-common) accepts only a single config, it seems like there's no way to pass more than one config anyway to backend-app-api, is there? 

In our team, we have been looking in moving Backstage to a combination of AWS Secrets Manager and AWS Parameter Store as configuration sources hence are interested in having two data sources. If there's no way to do that, we can definitely find a way to survive with one :)

Comment (freben) [reply packages/backend-common/src/config.ts]: Yeah it's been released as part of the `-next` line but not mainline. Alright I allowed it to break now then, @Rugvip check it out again to see if it seems alright!

**Review by ilya-kochetkov-cko — APPROVED** [review]

File: .changeset/sour-cooks-repeat.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'@backstage/backend-common': patch
+---
+
+Renamed the `loadBackendConfig` option `additionalConfig` to `additionalConfigs` as an array, and ensured that they get passed on properly.
+
+This is technically breaking, but the [original addition](https://github.com/backstage/backstage/pull/16643) hasn't been released in mainline yet so we are taking this step now as a `patch` change.
```
Comment (jhaals) [inline .changeset/sour-cooks-repeat.md:7]: Could we change the existing changeset in the PR mentioned for tomorrows release and skip this file entirely?

**Review by jhaals — APPROVED** [review]

File: .changeset/sour-cooks-repeat.md (hunk 2 above)
Comment (freben) [reply .changeset/sour-cooks-repeat.md:7]: Is that kosher after -next releases have been out? Maybe?

File: packages/backend-common/src/config.ts (hunk 1 above)
Comment (RobotSail) [reply packages/backend-common/src/config.ts]: @ilya-kochetkov-cko None of my code is holy, you can delete it for all I care. As long as the end result is the best solution to the problem is all I care about.

**Pull request merged**